wellcomeml/ml/cnn.py

```diff
@@ -0,0 +1,68 @@
+"""
+CNN architecture inspired from spacy for NLP tasks
+
+It follows the embed, encode, attend, predict framework
+Embed: learns a new embedding but can receive
+    pre trained embeddings as well
+Encode: stacked CNN with context window 3 that maintains
+    the size of input and applies dropout and layer norm
+Attend: not yet implemented
+Predict: softmax or sigmoid depending on number of outputs
+    and whether task is multilabel
+"""
+from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.metrics import f1_score
+import tensorflow as tf
+
+
+class CNNClassifier(BaseEstimator, ClassifierMixin):
+    def __init__(self, context_window = 3, learning_rate=0.001, batch_size=32, nb_epochs=5, dropout=0.2, multilabel=False):
+        self.context_window = context_window
+        self.learning_rate = learning_rate
+        self.batch_size = batch_size
+        self.nb_epochs = nb_epochs
+        self.dropout = dropout
+        self.multilabel = multilabel
+
+    def fit(self, X, Y, embedding_matrix=None):
+        sequence_length = X.shape[1]
+        vocab_size = X.max() + 1
+        emb_dim = embedding_matrix.shape[1] if embedding_matrix else 100
+        nb_outputs = Y.max() if not self.multilabel else Y.shape[1]
+
+        def residual_conv_block(x1):
+            filters = x1.shape[2]
+            x2 = tf.keras.layers.Conv1D(filters, self.context_window, padding='same', activation='relu')(x1)
+            x2 = tf.keras.layers.Dropout(self.dropout)(x2)
+            x2 = tf.keras.layers.LayerNormalization()(x2)
+            return tf.keras.layers.add([x1, x2])
+
+        embeddings_initializer = tf.keras.initializers.Constant(embedding_matrix) if embedding_matrix else 'uniform'
```

Comment:
Nice - so we can initialised it with pre-trained embeddings as well?

Reply:
yep, even though not documented well it is possible.